packages/react/src/components/Dropdown/DropdownSelectedItem.tsx

```diff
@@ -125,28 +126,39 @@ class DropdownSelectedItem extends UIComponent<WithAsProp<DropdownSelectedItemPr
   }: RenderResultConfig<DropdownSelectedItemProps>) {
     const { active, header, icon, image } = this.props
 
-    const iconElement = Icon.create(icon, {
-      defaultProps: {
-        'aria-label': `Remove ${header} from selection.`, // TODO: Extract this in a behaviour.
-        className: DropdownSelectedItem.slotClassNames.icon,
-        styles: styles.icon,
-      },
-      overrideProps: this.handleIconOverrides,
-    })
-    const imageElement = Image.create(image, {
-      defaultProps: {
-        avatar: true,
-        className: DropdownSelectedItem.slotClassNames.image,
-        styles: styles.image,
-      },
-    })
     const contentElement = Box.create(header, {
       defaultProps: {
         as: 'span',
         className: DropdownSelectedItem.slotClassNames.header,
         styles: styles.header,
       },
     })
+    const renderIcon = _.isNil(icon)
+      ? icon
+      : render =>
+          render(image, (ComponentType, props) =>
+            Icon.create(icon, {
+              defaultProps: {
+                'aria-label': `Remove ${header} from selection.`, // TODO: Extract this in a behaviour.
+                className: DropdownSelectedItem.slotClassNames.icon,
+                styles: styles.icon,
+              },
+              overrideProps: this.handleIconOverrides(props),
+            }),
+          )
+    const renderImage = _.isNil(image)
+      ? image
+      : render =>
+          render(image, (ComponentType, props) =>
+            Image.create(image, {
+              defaultProps: {
+                avatar: true,
+                className: DropdownSelectedItem.slotClassNames.image,
+                styles: styles.image,
+              },
+              overrideProps: props,
+            }),
+          )
```

Comment:
I am still disappointed with this.

As I see we used our own components in the wrong way, I expect that we can reuse `Image` and `Icon` components, but should we reuse `Label` component in this case? Not sure 🌵

Reply:
Is this the only place where we implicitly pass shorthand of a shorthand? This will be issue in every case we have like this. Meaning, we have shorthand for something that is shorthand in another element..

Reply:
We have this only in `Dropdown` components. I feel that it will be easier to have some custom components instead of this composition